extensions/mssql/src/objectManagement/ui/restoreDatabaseDialog.ts

```diff
@@ -462,16 +482,48 @@ export class RestoreDatabaseDialog extends ObjectManagementDialogBase<Database,
 	private async createBackupUrlFileBrowser(): Promise<void> {
 		let backupPath = await azdata.window.openBackupUrlBrowserDialog(this.options.connectionUri, '', true);
 		if (backupPath && !backupPath.includes('undefined')) {
-			this.backupFilePathInput.value = this.backupURLPath = backupPath;
+			this.backupURLPath = backupPath;
+			await this.backupFilePathInput.updateProperties({
+				value: this.backupURLPath
+			});
 		}
 	}
 
+	/**
+	 * Creates a file browser and sets the path to the backup url Path
+	 */
+	private async createRestoreS3Url(): Promise<void> {
+		const dialog = new S3CredentialsDialog();
```

Comment:
Are you prompting the user for a new credential each time? What if they want to select an existing credential they previously created?

Reply:
they can select it:
<img width="845" alt="image" src="https://github.com/microsoft/azuredatastudio/assets/34872381/1892005d-952d-478e-ae0e-b3db3d7401c1">

Reply:
Can you capture a gif showing the full experience - both for creating a new credential and for re-using an existing one? (and preferably attach it to the description)

Reply:
I updated the pr description to include the gifs